test/ad.jl

```diff
@@ -184,6 +229,27 @@ Vlist = ((ℂ^2, (ℂ^3)', ℂ^3, ℂ^2, (ℂ^2)'),
         test_rrule(LinearAlgebra.dot, A, B)
     end
 
+    @timedtestset "Matrix functions ($T)" for T in (Float64, ComplexF64)
+        for f in (sqrt, exp)
+            check_inferred = false # !(T <: Real) # not type-stable for real functions
+            t1 = randn(T, V[1] ← V[1])
+            t2 = randn(T, V[2] ← V[2])
+            d = DiagonalTensorMap{T}(undef, V[1])
+            randn!(d.data)
+            if T <: Real
+                d.data .= abs.(d.data)
```

Comment:
I assume this is only necessary for `f === sqrt`? Alternatively, you could write
```
logdata = randn(T, rdim(V[1])
d = DiagonalTensorMap(exp.(logdata), V[1])
```
This will create a purely positive `DiagonalTensorMap` in the real case.

Reply:
Just realized that there literally is a `randexp` function to do that in one go... Thanks for the suggestion, I'll change that.

Reply:
I don't think `randexp` works with complex numbers.

Reply:
But for complex numbers I don't need them to be positive, so I can just use `randn!`. I can definitely change it if you want, anything goes I guess?